main.go

```diff
@@ -85,39 +76,39 @@ func main() {
 	}
 
 	ctx = prefixcollector.WithKubernetesInterface(ctx, kubernetes.Interface(clientSet))
-	notifyChan := make(chan struct{}, 1)
 
-	currentNamespace, err := ioutil.ReadFile(currentNamespacePath)
+	prefixesOutputOption, err := getPrefixesOutputOption(config)
 	if err != nil {
-		span.Logger().Fatalf("Error reading namespace from secret: %v", err)
+		span.Logger().Fatal(err)
 	}
 
-	excludePrefixService := prefixcollector.NewExcludePrefixCollector(
-		notifyChan,
-		config.NSMConfigMapName,
-		strings.TrimSpace(string(currentNamespace)),
-		prefixsource.NewEnvPrefixSource(envPrefixes),
-		prefixsource.NewKubeAdmPrefixSource(ctx, notifyChan),
-		prefixsource.NewKubernetesPrefixSource(ctx, notifyChan),
-		prefixsource.NewConfigMapPrefixSource(ctx, notifyChan, config.ConfigMapName, config.ConfigMapNamespace),
+	notifyChan := make(chan struct{}, 1)
+	prefixCollector := prefixcollector.NewExcludePrefixCollector(
+		prefixesOutputOption,
+		prefixcollector.WithNotifyChan(notifyChan),
+		prefixcollector.WithSources(
+			prefixsource.NewEnvPrefixSource(config.ExcludedPrefixes),
+			prefixsource.NewKubeAdmPrefixSource(ctx, notifyChan),
+			prefixsource.NewKubernetesPrefixSource(ctx, notifyChan),
+			prefixsource.NewConfigMapPrefixSource(ctx, notifyChan, config.ConfigMapName, config.ConfigMapNamespace),
+		),
 	)
 
-	go excludePrefixService.Serve(ctx)
+	go prefixCollector.Serve(ctx)
 
 	span.Finish() // exclude main cycle run time from span timing
 	<-ctx.Done()
 }
 
-// validatedPrefixes returns list of validated via CIDR notation parsing prefixes
-func validatedPrefixes(prefixes []string) ([]string, error) {
-	var validatedPrefixes []string
-	for _, prefix := range prefixes {
-		_, _, err := net.ParseCIDR(prefix)
+func getPrefixesOutputOption(config *prefixcollector.Config) (prefixcollector.Option, error) {
+	if config.PrefixesOutputType == prefixcollector.ConfigMapOutputType {
```

Comment:
Can we move this condition into main func?
```suggestion
```

Reply:
Moved